card/lib/card/subcards.rb

```diff
@@ -228,7 +228,7 @@ def new_by_attributes name, attributes={}
     end
 
     def absolutize_subcard_name name
-      if @context_card.name =~ /^\+/
+      if @context_card.name =~ /^\+/ || name.blank?
```

Comment:
If you want to use a autoname rule for a subcard. For example in wikirate we add a source card as follows:
`add_subcard '', type_id: SourceID`.  Without the fix above the source card gets the same name as its supercard and the autoname rule is not used because the name is not empty. I think we should fix that in smartname. Or do you have reasion why an empty name should absolutize to the context name?